### DISCUSSION TITLE
Is it possible to run Cypress tests from a different repository?

### DISCUSSION THREAD

**Discussion by GeorgeXCV** [cypress-io/cypress#25562] [Questions and Help]
I have Cypress tests in a dedicated repo. The front-end code is in a different repo. When a developer opens a Pull Request I want them to have the option to trigger a subset of tests. Is this possible? As I'm unable to find any examples of this.

**Comment by portyu9** [comment]
Hi @GeorgeXCV 

Yes. Cypress does not require the application code and Cypress specs to be in the same Git repository.

The repository orchestration is a CI concern. A typical PR workflow is:

1. Check out the front-end repository at the PR commit.
2. Check out the Cypress repository as well.
3. Start/build the front-end application.
4. Run Cypress from the test repository against that application's URL.
5. Use `--spec` to select the requested subset.

For example:

```bash
npx cypress run \
  --config baseUrl=http://localhost:3000 \
  --spec 'cypress/e2e/smoke/**/*.cy.ts'
```

`--spec` paths are relative to the Cypress project and must also match its configured `specPattern`.

Another valid architecture is to have the front-end PR workflow trigger a workflow/pipeline in the Cypress repository, passing:

- the front-end repository
- the PR commit SHA
- the requested test group/spec pattern

The Cypress workflow can then check out that exact front-end commit, start it, and execute the selected tests.

The main thing to avoid is testing whatever happens to be the latest `main` branch of the front-end repository. The Cypress run should use the **exact PR commit SHA** so the result corresponds to the code being reviewed.

For private repositories, the CI credential/token also needs read access to both repositories.

Please mark this answer as accepted if it helped, thank you.
